### DISCUSSION TITLE
使用rsbuild报错Not allowed to load local resource

### DISCUSSION THREAD

**Discussion by wChenonly** [web-infra-dev/rsbuild#1323] [Q&A] [answered]
![image](https://github.com/web-infra-dev/rsbuild/assets/37658262/265abed4-a1ce-4ede-959f-2aec93257290)

**Comment by wChenonly** [comment]
![image](https://github.com/web-infra-dev/rsbuild/assets/37658262/2c9775e5-a8c9-4809-a3c1-c753a2ce0ae2)

**Comment by chenjiahan** [comment]
来个复现仓库~

**Comment by wChenonly** [comment]
> 来个复现仓库~
https://github.com/wChenonly/rsbuild-Not-allowed-to-load

**Comment by chenjiahan** [comment, marked as answer]
通过 new URL(url, import.meta.url) 拿到的是一个 file 协议地址，Chrome 应该是不允许加载的，参考 https://support.google.com/chrome/thread/162698059/not-allowed-to-load-local-resource-how-to-fix-this?hl=en

**Comment by wChenonly** [comment]
试了下，新建一个vite-demo引入库开发的时候是可以的，可能跟 这个有关https://vitejs.dev/config/server-options.html#server-fs-allow
但是打包后是不行的，
vuecli和rsbuild的行为是一致的，进一步发现是vite在开发库的时候，在lib模式下，并没有把资源内连，测试仓库https://github.com/wChenonly/vite-lib-assetsInlineLimit
<img width="702" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/37658262/02154a2e-bf95-4a40-bc6e-3f073cce7e39">
https://cn.vitejs.dev/config/build-options.html#build-assetsinlinelimit

**Comment by wChenonly** [comment]
> 还有一个问题，我基于vant二次封装的slider组件，在低版本的浏览器会报错 ?? 语法。 <img alt="image" width="1507" src="https://private-user-images.githubusercontent.com/37658262/298013227-c83d721e-164d-4bc1-86a2-205d0e9b28e3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2NTY1NDksIm5iZiI6MTcwNTY1NjI0OSwicGF0aCI6Ii8zNzY1ODI2Mi8yOTgwMTMyMjctYzgzZDcyMWUtMTY0ZC00YmMxLTg2YTItMjA1ZDBlOWIyOGUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDA5MjQwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2ZTlmYzViOGJmZjVkZDJhMDE2N2MxNjUzNTg0ODEzMmY5N2QyNjVmYTlmZTkyNDVmMzgwZjQ0ODUxZGVmZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kojHKiohfTd17KRWEm_Alt0KR6jFI0yIpGWqPdjsutc">
> 
> 但是我看vant源码打包以后确实是转换了的 ![image](https://private-user-images.githubusercontent.com/37658262/298013157-4ae236e9-ef19-4276-b408-1a33d876518e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2NTY1NDksIm5iZiI6MTcwNTY1NjI0OSwicGF0aCI6Ii8zNzY1ODI2Mi8yOTgwMTMxNTctNGFlMjM2ZTktZWYxOS00Mjc2LWI0MDgtMWEzM2Q4NzY1MThlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDA5MjQwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmZDM2OTllZjZjNzBkYzExYzJlZjIwMmFjYjlmYTlmOGI3OTkzYzAxN2UzNTAzNGIyZDgyODJjM2VjMTI4NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rcXrmuHx2EZoI1OboqNaBRctccdUljdhxyPacRPLaiA)
> 
> 但是我打包以后，就发现这个 ?? 语法并没有转换 ![image](https://private-user-images.githubusercontent.com/37658262/298014336-bcfcab10-d05e-485d-bf88-87d46cea6b48.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2NTY1NDksIm5iZiI6MTcwNTY1NjI0OSwicGF0aCI6Ii8zNzY1ODI2Mi8yOTgwMTQzMzYtYmNmY2FiMTAtZDA1ZS00ODVkLWJmODgtODdkNDZjZWE2YjQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDA5MjQwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA4MjQ5ZDk3NTlkYTBlY2Y5NDIyOWU4ZTViZjMxNDViYzA4ZjMxMzZiMmNiYjZhNmMxNzk2YmMwYTk5MGJlMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aJ00hMl42X-WscH-UodLN5rB_ux3BvTVvP7IcUoct5Y)
> 
> 我不确定是不是我打包的问题
debug了下，没找到原因🤣,还是之前的代码仓

**Reply by chenjiahan** [reply]
是不是 w-ui 这个项目的打包问题呢，我看是 vite 打包的

**Reply by wChenonly** [reply]
我再试试看，是vite打包的。
